bin/tests/integration/named_metrics_tests.rs

```diff
@@ -0,0 +1,260 @@
+                &store_file_dynamic_updated,
+                Some(0f64),
+            );
+        }
+    })
+}
+
+async fn fetch_parse_check_metrics(socket_ports: &SocketPorts) -> Scrape {
+    let prometheus_port = socket_ports.get_v4(ServerProtocol::PrometheusMetrics);
+    let addr = SocketAddr::from((
+        Ipv4Addr::LOCALHOST,
+        prometheus_port.expect("no prometheus_port"),
+    ));
+
+    // the collect interval for process metrics is set to 1000ms
+    // wait to avoid missing the process metrics
+    tokio::time::sleep(Duration::from_millis(1000)).await;
+
+    // fetch from metrics from server
+    let body = reqwest::get(format!("http://{}:{}/", addr.ip(), addr.port()))
+        .await
+        .unwrap()
+        .text()
+        .await
+        .unwrap();
+    let lines: Vec<_> = body.lines().map(|s| Ok(s.to_owned())).collect();
+    let scrape = Scrape::parse(lines.into_iter()).unwrap();
+
+    test_metrics_description_present(scrape.clone());
+    scrape
+}
+
+fn test_metrics_description_present(scrape: Scrape) {
+    let missing: Vec<String> = scrape
+        .samples
+        .into_iter()
+        .filter_map(|s| {
+            if !scrape.docs.contains_key(&s.metric) {
+                Some(s.metric)
+            } else {
+                None
+            }
+        })
+        .collect();
+
+    assert_eq!(
+        Vec::<String>::new(),
+        missing,
+        "due to missing metric descriptions"
+    );
+}
+
+fn verify_metric(metrics: &Scrape, name: &str, labels: &[(&str, &str)], value: Option<f64>) {
+    let named: Vec<_> = metrics
+        .samples
+        .iter()
+        .filter(|s| s.metric == name)
+        .collect();
+
+    assert!(!named.is_empty(), "failed locating metric: {}", name);
+
+    let labeled: Vec<_> = named
+        .iter()
+        .filter(|s| {
+            labels.len()
+                == labels
+                    .iter()
+                    .filter(|(k, v)| {
+                        let label = s.labels.get(k);
+                        label.is_some() && label.unwrap() == *v
+                    })
+                    .count()
+        })
+        .collect();
+
+    assert_eq!(
+        labeled.len(),
+        1,
+        "locating metric: {}, labels: {:?}",
+        name,
+        labels
+    );
+
+    if let Some(value) = value {
+        labeled.iter().for_each(|s| match s.value {
+            Value::Counter(v) | Value::Gauge(v) | Value::Untyped(v) => {
+                println!(
+                    "checking metric: {}, labels: {}, value: {}",
+                    s.metric, s.labels, v
+                );
+                assert_eq!(
+                    v, value,
+                    "checking metric: {}, labels: {}, value: {}",
+                    s.metric, s.labels, v
+                )
+            }
+            Value::Histogram(_) | Value::Summary(_) => {
+                panic!("metric type check not supported")
+            }
+        })
+    }
```

Comment:
Do a let-else to reduce rightward drift?

Reply:
Done.